### PR TITLE
Adv embedding explanations

### PR DESCRIPTION
**Pull request by jambran**
Explanations for 
- Word2Vec
- FastText
- GloVe
- BERT

Implementation in python using gensim for 
- Word2Vec
- FastText

Tutorials for implementing the other two linked, but not worth it to dive deep on. 

It's been a hot minute since I've looked into GloVe & BERT - did a lot of linking and quoting, as they explain it way better than I can. Let me know if it needs more images, anything else to make it clear. Ton of info.

### CODE REVIEW

File: word_embedding_advanced_tutorial.ipynb [hunk 1]
```diff
@@ -352,7 +376,292 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### FastText\n",
+    "\n",
+    "The reason FastText achieves such incredible performance for word representations and sentence classification is in-part due it's use of character level information.\n",
+    "\n",
+    "From [FastText: Under the Hood](https://towardsdatascience.com/fasttext-under-the-hood-11efc57b2b3)\n",
+    ">Each word is represented as a bag of character n-grams in addition to the word itself, so for example, for the word matter, with n = 3, the fastText representations for the character n-grams is <ma, mat, att, tte, ter, er>. < and > are added as boundary symbols to distinguish the ngram of a word from a word itself, so for example, if the word mat is part of the vocabulary, it is represented as <mat>. \n",
```
Comment (kirsten-stallings) [inline word_embedding_advanced_tutorial.ipynb]: for the word "matter"

Comment (kirsten-stallings) [reply word_embedding_advanced_tutorial.ipynb]: We are inconsistent between capitalizing FastText and not

File: word_embedding_advanced_tutorial.ipynb [hunk 2]
```diff
@@ -352,7 +376,292 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### GloVe\n",
+    "\n",
+    "From [the GloVe website](https://nlp.stanford.edu/projects/glove/)\n",
+    ">The GloVe model is trained on the non-zero entries of a global word-word co-occurrence matrix, which tabulates how frequently words co-occur with one another in a given corpus. Populating this matrix requires a single pass through the entire corpus to collect the statistics. For large corpora, this pass can be computationally expensive, but it is a one-time up-front cost. Subsequent training iterations are much faster because the number of non-zero matrix entries is typically much smaller than the total number of words in the corpus.\n",
```
Comment (kirsten-stallings) [inline word_embedding_advanced_tutorial.ipynb]: a link explaining co-occurance matrix? I feel like some people might not know what that is but also not ask

Comment (kirsten-stallings) [reply word_embedding_advanced_tutorial.ipynb]: or a picture

File: word_embedding_advanced_tutorial.ipynb (hunk 1 above)
Comment (jambran) [reply word_embedding_advanced_tutorial.ipynb]: Good call out, this one's a quote from the article though. I'll adjust it though

File: word_embedding_advanced_tutorial.ipynb (hunk 2 above)
Comment (jambran) [reply word_embedding_advanced_tutorial.ipynb]: Really good idea, thank you!

**Pull request merged**